src/resources/transactions/transaction.ts

```diff
@@ -46,7 +46,7 @@ export const fetchTransaction = async ({
     });
 
     const tx = response?.data?.payload?.transaction;
-    if (!tx) {
+    if (!tx || !tx?.status || (tx?.status as string) === '') {
       return null;
```

Comment:
when we return `null`, it just gets merged into the existing pending transaction object as : 
```{ ...existingTxn, ...null } ``` and doesn't change anything, so it doesn't get filtered out from the pending txn list; basically gets treated the same as "backend doesn't know about this txn yet"